tests/QueryWithFiltersTest.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Data\Db\Tests;
+
+use DateTime;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Data\Db\Filter\Between as FilterBetween;
+use Yiisoft\Data\Db\Filter\CompareFilter;
+use Yiisoft\Data\Db\Filter\Equals as FilterEquals;
+use Yiisoft\Data\Db\Filter\GreaterThan as FilterGreaterThan;
+use Yiisoft\Data\Db\Filter\GreaterThanOrEqual as FilterGreaterThanOrEqual;
+use Yiisoft\Data\Db\Filter\In as FilterIn;
+use Yiisoft\Data\Db\Filter\LessThan as FilterLessThan;
+use Yiisoft\Data\Db\Filter\LessThanOrEqual as FilterLessThanOrEqual;
+use Yiisoft\Data\Db\Filter\Like as FilterLike;
+use Yiisoft\Data\Db\Filter\NotEquals as FilterNotEquals;
+use Yiisoft\Data\Db\Filter\OrLike as FilterOrLike;
+use Yiisoft\Data\Db\QueryDataReader;
+use Yiisoft\Data\Db\Tests\Support\TestTrait;
+use Yiisoft\Db\Expression\Expression;
+use Yiisoft\Db\Query\Query;
+
+final class QueryWithFiltersTest extends TestCase
+{
+    use TestTrait;
+
+    public static function setUpBeforeClass(): void
+    {
+        CompareFilter::$mainDateTimeFormat = 'Y-m-d H:i:s';
+    }
+
+    public function simpleDataProvider(): array
+    {
+        return [
+            //EqualsHandler
+            [
+                new FilterEquals('equals', 1),
+                '[equals] = 1',
```

Comment:
Shouldn't it produce `[[equals]]`?

https://github.com/yiisoft/db/tree/master/docs/en#quote-table-and-column-names

Reply:
Also, how about prepared statements instead of values in the query directly?

Reply:
> Also, how about prepared statements instead of values in the query directly?

I will think about this. Current PR just for implements latests changes from depends packages